docs/src/main/asciidoc/doc-reference.adoc

```diff
@@ -289,10 +288,10 @@ xref:<anchor-ID>[]
 +
 [source,asciidoc]
 --
-xref:<target-file>.adoc#<anchord-ID>[]
+xref:<target-file>.adoc#<anchord-ID>[<anchor-name>]
 --
 
-NOTE: The `[]` part of your `xref` anchors sets up a desired anchor label in cases where the default label does not fit the content.
+NOTE: The `[]` part of your `xref` anchors sets up a desired anchor label in cases where the default label does not fit the content. However, always specifying the desired anchor or `xref` name.
```

Comment:
I agree with @maxandersen. It would be great to resolve this automatically because not all contributors read the guides and we have a few more issues like this in the PV tooling system.

However, if we don't find a smart way to resolve this, here's a minor tweak for clarity:

```suggestion
Important: To ensure that the inline anchor link renders correctly,  always populate the anchor label `[]` part of the`xref` with the correct text that fits the sentence.

Reply:
Thank you for the suggestion. I am more for the original line because it is not an upstream issue they won't render elsewhere. That's why I wanted to say that they should specify them manually and check if the context matches its purpose and the displayed xref output fits the sentence, and it makes sense (looks good) overall. Or, ofc, keep it empty in case the heading that is automagically sourced there is great and rendered output of it fits the context and looks good in a sentence (and not causing an issue in other repo). :)

Reply:
Something about these lines?

Important: The `[]` part of your `xref` anchors sets up a desired anchor name when the default name, the value of which `asciidoc` sources from chapter headings, file names, or anchors, does not fit in the current context. Always specify the intended anchor or `xref` name and verify if the displayed name fits the context of your content.